code/datums/components/shuttle_cling.dm

```diff
@@ -97,8 +101,13 @@
 		return ALL_GOOD
 
 	if(!isliving(movee))
+		var/datum/component/singularity/singularity = movee.GetComponent(/datum/component/singularity)
+		if(singularity) // nothing can block the singularity
+			return SUPER_NOT_HOLDING_ON
```

Comment:
Could you add a trait for this instead? I don't enjoy typechecks or GetComponent when they can be avoided

Reply:
Applied with https://github.com/tgstation/tgstation/pull/77188/commits/4cf8865d55796735fcdec391ff6b42039428a19a